internal/query/mock/mock.go

```diff
@@ -71,8 +76,28 @@ func (e *Engine) DeleteRelationships(ctx context.Context, resource types.Resourc
 
 // NewResourceFromID creates a new resource object based on the given ID.
 func (e *Engine) NewResourceFromID(id gidx.PrefixedID) (types.Resource, error) {
+	prefix := id.Prefix()
+
+	var rType *types.ResourceType
+
+	if e.schema == nil {
+		e.schema = iapl.DefaultPolicy().Schema()
```

Comment:
Feels like the schema should be preloaded and this should be an error?

Reply:
This is just for the mock. Most tests just define an empty mock Engine. So this simplifies to use the default schema if we aren't testing something specific.